internal/http/client.go

```diff
@@ -210,6 +216,25 @@ func OptSetLogger(log log.Modular) func(*Client) {
 	}
 }
 
+// OptSetMultiPart sets the multipart to request.
+func OptSetMultiPart(multipart []client.Part) func(*Client) {
+	return func(t *Client) {
+		if len(multipart) > 0 {
+			for _, v := range multipart {
+				data, _ := interop.NewBloblangField(t.mgr, v.Data)
```

Comment:
The error handling here really needs to be exposed somehow. Ideally the component that is calling `OptSetMultipart` should be calling `NewBloblangField` itself so that it can properly return errors, and then `client.Part` comes with the `*field.Expression` values.

Reply:
Ideally, this should happen in internal HTTP's NewClient method like we are handling for other fields, but as HTTP client doesn't have access to MultiPart because this is like an extra child to HTTP client as this belongs to only output processor only.
So NewClient's definition will change, somehow like this so that it got access to MultiParts and we do the error handling too.

Change purposing to an existing function definition -
 ```func NewClient(conf client.Config, parts []client.Part, opts ...func(*Client)) (*Client, error) {}```

original - ```func NewClient(conf client.Config, opts ...func(*Client)) (*Client, error) {}```
If this sounds okay to you, should I go ahead with the code change? I will cover unit test cases also.

Reply:
Hey @YachikaRalhan, I've implemented what I was describing on this branch: https://github.com/Jeffail/benthos/tree/YachikaRalhan-http-multipart-request, I've also renamed the `data` field to `body`. I think with those changes I'm happy to merge, would you be able to rebase this PR from that branch and add some unit tests?

Reply:
Closing this PR, as I have created a new [PR](https://github.com/Jeffail/benthos/pull/1087) @Jeffail